clang/lib/CodeGen/CGBuiltin.cpp

```diff
@@ -21866,6 +21933,117 @@ RValue CodeGenFunction::EmitIntelFPGAMemBuiltin(const CallExpr *E) {
   return RValue::get(Ann);
 }
 
+llvm::CallInst *CodeGenFunction::EmitFPBuiltinIndirectCall(
+    llvm::FunctionType *IRFuncTy, const SmallVectorImpl<llvm::Value *> &IRArgs,
+    llvm::Value *FnPtr, const FunctionDecl *FD) {
+  llvm::Function *Func;
+  unsigned FPAccuracyIntrinsicID = 0;
+  StringRef Name;
+  if (CurrentBuiltinID == 0) {
+    // Even if the current function doesn't have a clang builtin, create
+    // an 'fpbuiltin-max-error' attribute for it; unless it's marked with
+    // an NoBuiltin attribute.
+    if (!FD->hasAttr<NoBuiltinAttr>()) {
+      Name = FD->getName();
+      FPAccuracyIntrinsicID =
+          llvm::StringSwitch<unsigned>(Name)
+              .Case("fadd", llvm::Intrinsic::fpbuiltin_fadd)
+              .Case("fdiv", llvm::Intrinsic::fpbuiltin_fdiv)
+              .Case("fmul", llvm::Intrinsic::fpbuiltin_fmul)
+              .Case("fsub", llvm::Intrinsic::fpbuiltin_fsub)
+              .Case("frem", llvm::Intrinsic::fpbuiltin_frem)
+              .Case("sincos", llvm::Intrinsic::fpbuiltin_sincos)
+              .Case("exp10", llvm::Intrinsic::fpbuiltin_exp10)
+              .Case("rsqrt", llvm::Intrinsic::fpbuiltin_rsqrt)
+              .Default(-1);
+      assert(FPAccuracyIntrinsicID != -1 && "unexpected fpbuiltin ID");
```

Comment:
```suggestion
      FPAccuracyIntrinsicID =
          llvm::StringSwitch<unsigned>(Name)
              .Case("fadd", llvm::Intrinsic::fpbuiltin_fadd)
              .Case("fdiv", llvm::Intrinsic::fpbuiltin_fdiv)
              .Case("fmul", llvm::Intrinsic::fpbuiltin_fmul)
              .Case("fsub", llvm::Intrinsic::fpbuiltin_fsub)
              .Case("frem", llvm::Intrinsic::fpbuiltin_frem)
              .Case("sincos", llvm::Intrinsic::fpbuiltin_sincos)
              .Case("exp10", llvm::Intrinsic::fpbuiltin_exp10)
              .Case("rsqrt", llvm::Intrinsic::fpbuiltin_rsqrt);
```
`StringSwitch` will assert if you fall off the end without a default, so this is equivalent.